src/database/database-common/src/db_provider.rs

```diff
@@ -0,0 +1,42 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+/////////////////////////////////////////////////////////////////////////////////////////
+
+#[derive(Debug)]
+pub enum DatabaseProvider {
+    Postgres,
+    MySql,
+    MariaDB,
```

Comment:
No strategy yet, since business requirements on this regards are undetermined. Remember we discussed Microsoft SQL Server support as well..

Yes, I would prefer to continously keep in mind that the solution should be database-agnostic, and keeping 2 or more lines would be helpful. It could be SQLite, or MySQL, or both .. Intuitively it feels like keeping them growing incrementally would be less painful than adding full database support for all repositories at once. On the other hand, we might never need those. 

I guess, business requirements have to be defined first.